app/controllers/spree/admin/orders_controller.rb

```diff
@@ -9,20 +9,31 @@ class OrdersController < Spree::Admin::BaseController
       helper CheckoutHelper
 
       before_action :load_order, only: [:edit, :update, :fire, :resend,
-                                        :invoice, :print]
-      before_action :load_distribution_choices, only: [:new, :edit, :update]
+                                        :invoice, :print, :set_distribution]
+      before_action :load_distribution_choices, only: [:new, :edit, :update, :set_distribution]
 
-      # Ensure that the distributor is set for an order when
-      before_action :ensure_distribution, only: :new
       before_action :require_distributor_abn, only: :invoice
 
       respond_to :html, :json
 
       def new
         @order = Order.create
         @order.created_by = spree_current_user
+        @order.generate_order_number
         @order.save
-        redirect_to spree.edit_admin_order_url(@order)
+        redirect_to spree.distribution_admin_order_path(@order)
+      end
+
+      def set_distribution
+        return if order_params.blank?
+
+        on_update
+
+        @order.update(order_params)
+        return unless @order.valid?(:set_distribution_step)
+
+        @order.save
```

Comment:
Now I see the additional validations. I think that you want to use `assign_attributes` to prevent saving before running those validations. You can also pass the context to `save`.
```suggestion
        @order.assign_attributes(order_params)
        return unless @order.save(context: :set_distribution_step)
```

This runs the validations only once.